Lesson 2 - Functions/code/homework/code.js

```diff
@@ -24,23 +24,56 @@
 
     global.UAM.addAircraft = function (newAircraftCode) {
         // function should return new aircraft object
+        var aircraftObj = {
+            code: newAircraftCode,
+            services: []
+        }
+        global.UAM.aircrafts.push(aircraftObj);
+        return aircraftObj;
+        
     };
 
     global.UAM.removeAircraft = function (aircraftObj) {
         // !!!
+        if( global.UAM.aircrafts.indexOf(aircraftObj)>=0){
+            global.UAM.aircrafts.splice( global.UAM.aircrafts.indexOf(aircraftObj), 1 );
+        }
+        
     };
 
     global.UAM.addWorkToAircraft = function(aircraftObj, name, timeToExxecute) {
         // !!!
+        aircraftObj.services.push({
+        name: name,
+        timeToExecute: timeToExxecute
+    });
     };
         
-    global.UAM.reduceTimeToExecute = function(time) {
+    global.UAM.reduceTimeToExecute = function(aircraftObj, time) {
         // !!!
-    };
-    
+        if( global.UAM.aircrafts.indexOf(aircraftObj)>=0){
+            if(aircraftObj.services !==null){
```

Comment:
You should avoid compare to `null` or `undefined` directly. Just variable name will do.
```javascript
if(!aircraftObj.services) {
 ...
```